components/ChallengeMaterial.tsx

```diff
@@ -82,23 +94,51 @@ export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
   )
 }
 
-const ChallengeMaterial: React.FC<ChallengeMaterialProps> = props => {
-  if (!props.challenges) {
+const ChallengeMaterial: React.FC<ChallengeMaterialProps> = ({
+  userSubmissions,
+  challenges
+}) => {
+  if (!challenges.length) {
     return <h1>No Challenges for this lesson</h1>
   }
-  const sortedChallenges: Challenge[] = props.challenges.sort(
+  //create an object to evaluate the student's status with a challenge
+  const userSubmissionsObject = userSubmissions.reduce(
+    (acc: UserSubmissionsObject, submission: UserSubmission) => {
+      acc[submission.challengeId] = submission
+      return acc
+    },
+    {}
+  )
+  //create a new Challenges array with user submission data integrated
+  const challengesWithSubmissionData = challenges.reduce(
+    (acc: ChallengeSubmissionData[], challenge: Challenge) => {
+      const submissionData = userSubmissionsObject[challenge.id] || {}
+      acc.push(
+        Object.assign(challenge, {
+          status: submissionData.status || 'unsubmitted',
+          submissionData
+        })
+      )
+      return acc
+    },
+    []
+  )
+  //sort challenges by order
+  const sortedChallenges: ChallengeSubmissionData[] = challengesWithSubmissionData.sort(
     (a, b) => a.order - b.order
   )
-  //TODO: once user has completed a challenge find the first incomplete challenge that
-  //hasn't been completed to set as default challengeID
-  //assign currentChallenge to first challenge if no match found
+
+  //find first challenge that is not passed
+  const firstChallenge = sortedChallenges.find(
+    (challenge: ChallengeSubmissionData) => challenge.status !== 'passed'
+  ) as ChallengeSubmissionData
+
   const [currentChallengeID, setCurrentChallenge] = useState<
     CurrentChallengeID
-  >()
-  const currentChallenge =
-    sortedChallenges.find(
-      (challenge: Challenge) => challenge.id === currentChallengeID
-    ) || sortedChallenges[0]
+  >(firstChallenge.id)
```

Comment:
this way is pretty inefficient, because `firstChallenge` is computed on every state change even though it is pretty unnecessary.... instead start this at null and add `if` condition.

```
...sortedChallenges.find( (challenge) => {
  if (currentChallengeId) return currentChallengeId === challenge.id
  return challenge.status !== 'passed'
})
```